src/utils/style-formatter.js

```diff
@@ -75,4 +73,22 @@ export default {
 
     return styles;
   },
+  setFontSizeAndFamily({ button, Theme, layout }) {
+    const { style } = layout;
+    function setFontsize(textElement, newFontsize) {
+      textElement.setAttribute(
+        'style',
+        `white-space: nowrap; font-size: ${newFontsize}px; font-family: ${colorUtils.getFontFamily(Theme)};`
+      );
+    }
+    const text = button.firstElementChild;
+    text.textContent = style.label;
+    setFontsize(text, button.clientHeight);
+    let newFontsize = (button.clientHeight / text.offsetHeight) * button.clientHeight;
+    setFontsize(text, newFontsize);
+    if (text.offsetWidth + 8 > button.clientWidth) {
+      newFontsize *= (button.clientWidth - 8) / text.offsetWidth;
+    }
+    setFontsize(text, (newFontsize * style.font.size) / 100);
```

Comment:
Set the font size three times?

Reply:
The calculation is done in three steps:
1. Set the font size to the height of the parent container
2. Adjust the font size based on the height ratio of the parent container and the text 
3. Adjust the font size if the text width is bigger than the parent width

Reply:
You need to actually set the font size in order to do the next calculation otherwise the size of the text box is not adjusted

Reply:
fair enough :)